### PR TITLE
[BACKLOG-9602][BACKLOG-10485] Merging properties into type.application …

### PR DESCRIPTION
**Pull request by davidmsantos90**
…instead of overriding it.

@pentaho/millenniumfalcon please review.

### CODE REVIEW

File: package-res/resources/web/pentaho/type/_type.js [hunk 1]
```diff
@@ -603,6 +607,26 @@ define([
       },
       // endregion
 
+      // region application property
+      _application: null,
```
Comment (dcleao) [inline package-res/resources/web/pentaho/type/_type.js]: Think this should be `{}` right from the top.

File: package-res/resources/web/pentaho/type/_type.js [hunk 2]
```diff
@@ -603,6 +607,26 @@ define([
...
+       * that can be modified via the [Configuration Service]{@link pentaho.type.config.ConfigurationService}.
+       *
+       * @type Object
+       */
+      get application() {
+        return this._application;
+      },
+
+      set application(value) {
+        typeUtil.mergeSpecs(this._application, value);
```
Comment (dcleao) [inline package-res/resources/web/pentaho/type/_type.js]: Think this should ignore a _nully_ `value` (and the docs should say that).

File: package-res/resources/web/pentaho/type/_type.js [hunk 3]
```diff
@@ -603,6 +607,26 @@ define([
...
+      // region application property
+      _application: null,
+
+      /**
+       * Gets or sets the `application` attribute of this type.
+       *
+       * The application property serves as a bag of properties specific to the _container application_
+       * that can be modified via the [Configuration Service]{@link pentaho.type.config.ConfigurationService}.
+       *
+       * @type Object
```
Comment (dcleao) [inline package-res/resources/web/pentaho/type/_type.js]: Change to `@type object` to clarify that it is never `null`.

File: package-res/resources/web/pentaho/type/_doc/spec/ITypeProto.jsdoc [hunk 4]
```diff
@@ -58,6 +58,17 @@
...
 
 /**
+ * The `application` property of the type.
+ *
+ * The application property serves as a bag of properties specific to the _container application_,
+ * that can be modified via the [Configuration Service]{@link pentaho.type.config.ConfigurationService}.
+ *
+ * @name application
+ * @memberOf pentaho.type.spec.ITypeProto#
+ * @type Object
```
Comment (dcleao) [inline package-res/resources/web/pentaho/type/_doc/spec/ITypeProto.jsdoc]: Add `@see pentaho.type.Type#application`.

**Comment by dcleao** [comment]
👍 👍 👍

File: package-res/resources/web/pentaho/type/_type.js (hunk 2 above)
Comment (davidmsantos90) [inline package-res/resources/web/pentaho/type/_type.js]: mergeSpecs is already ignoring nully values. updated docs

**Comment by davidmsantos90** [comment]
@pentaho/millenniumfalcon pr updated with the reviews implemented. please review.

**Pull request merged**